src/agent/onefuzz-agent/src/tasks/report/crash_report.rs

```diff
@@ -139,9 +149,57 @@ impl CrashTestResult {
         }
         Ok(())
     }
+
+    pub async fn save_regression(
+        self,
+        original_report: Option<CrashReport>,
+        reports: &Option<SyncedDir>,
+        no_repro: &Option<SyncedDir>,
+        prefix: impl AsRef<str>,
+    ) -> Result<()> {
+        match self {
+            Self::CrashReport(report) => {
+                // Use SHA-256 of call stack as dedupe key.
+                if let Some(reports) = reports {
+                    let name = format!("{}{}", prefix.as_ref(), report.unique_blob_name());
+                    if upload_or_save_local(
+                        &RegressionReport {
+                            crash_result: Self::CrashReport(report),
+                            original_report,
+                        },
+                        &name,
+                        reports,
+                    )
+                    .await?
+                    {
```

Comment:
Is this block necessary for lifetime reasons?

Reply:
this is the code block for the if clause on line 165

Reply:
When I first read it, I wondered the same thing.  Perhaps making RegressionReport a named variable would make this easier to understand.

Reply:
Oh wow, yeah. I totally misread that!

Stylistically, I suggest as a team that we try to avoid pattern matching again complex RHS expressions. It fares particularly badly when combined with `rustfmt`.

Reply:
Can we customize the rustfmt instead ? may just increase the max line length. 
The idea changing the way we write our code because of a tool that is supposed to help not think about formatting seems weird to me.

Reply:
> The idea changing the way we write our code because of a tool that is supposed to help not think about formatting seems weird to me.

I agree, we should avoid doing that! The deeper reason is: I often find judiciously-named temporaries to be helpful for keeping less stuff in my head at one time. This is especially true if the temporary is going to be used in a conditional or RHS of an `if let`/`match`. This varies from case to case, of course, and I don't suggest we should be rigid / make a strict rule about it.